src/main/java/org/edgexfoundry/messaging/MQTTSender.java

```diff
@@ -97,6 +101,16 @@ private void connectClient() {
 			connOpts.setPassword(password.toCharArray());
 			connOpts.setCleanSession(true);
 			connOpts.setKeepAliveInterval(keepAlive);
+			if (secureConnection) {
+				try {
+					EdgeXSecurityManager secMgr = new EdgeXSecurityManager();
+					SSLContext ctxt = SSLContext.getInstance("TLS");
+					ctxt.init(secMgr.getKeyManagers(), secMgr.getTrustManagers(), new java.security.SecureRandom());
+					connOpts.setSocketFactory(ctxt.getSocketFactory());
+				} catch (Exception e) {
+					logger.debug("Failed to initialize secure connection to broker.  Will attempt unsecure connection.", e);
```

Comment:
Is this our desired failure behavior when attempting an MQTTS connection? I can see arguments in both directions (security vs. functionality), but perhaps this needs to be configurable with some sort of strictness check (@janko-isidorovic @jpwhitemn)?

Regardless, log level for this sort of failure should be bumped up to the `error` level.